### ISSUE TITLE
Usability issues: first user experience report

### ISSUE THREAD

**Issue by axelsimon** [perkeep/perkeep#1244]
HI there,

I finally got around to giving Perkeep a shot, as it had been on my radar for a while and love its goals and its starting principles.

But, and it pains me a bit to say this, it's basically unusable for non-developpers (and I say this as someone who has been self-hosting for years).

This issue has turned into a lengthy first-use experience report. I share it with you in the hopes that a set a fresh eyes can point out or remind you of some issues that affect Perkeep.

## Context
I'm approaching Perkeep as someone who would want to run an instance for themselves and their family and be able to confidently recommend it to them. I'm not entirely sure who Perkeep is designed with in mind, so some of the next comments might not be relevant. Nevertheless, I thought they might be useful to share, in the spirit of improving FOSS for everyday people.

## Setup
I'm running Perkeep on a server on my local network (it's not, but it could be a raspberry pi, which is a stated target/option in the documentation) and browsing to it from a laptop, which believe to be a rather common scenario.

This is an old box running Debian 10 (testing, aka buster) with a recent kernel (4.18.03). I'm reverse proxying through nginx (but my first tests where without reverse proxying).

### Documentation
I found myself going in circles from A to B and back to A quite often reading the documentation.

An example: https://perkeep.org/download#getting-started tells you how to start perkeep and that it should open a browser window. It doesn't tell you to expect the window to ask you for a user and password and what these should be (a default one, maybe perkeep/perkeep, something else?). It links to https://perkeep.org/doc/server-config which is already very technical for a general audience just trying to simply give Perkeep a go, and doesn't really answer that question ("localhost: Accept connections coming from localhost. On Linux, this means connections from localhost that are also from the same user as the user running the server.": wait, I'm not connecting from localhost, so what *is* the password?). I find myself still not knowing what user and password to put in, as my case (connecting from a laptop to a server on the same local network) isn't really considered in the documentation. 

Wait, maybe I haven't read the docs properly. Oh, this says "getting started", it will help me. Oh no, it just takes me back to how to start perkeepd, not what to do immediately after, once it's running.

Overall, this happens often: the documentation goes from "here's how you start it" to "here's how you configure some advanced thing" with not much in the middle.

(In my case I first used a socks5 SSH tunnel to my server to browse as if locally, then added a user:password to `server-config.json`, but to me this is already too far down the rabbit hole **before I've even seen the UI**. )

### Web UI
The web UI is confusing and unfinished, but I can imagine you are aware of this fact.

First of all, the main screen is bare, except for a menu at the top left and a search and map button at the top right. While this is a clean and good start, there is nothing on the screen to help a new user.
What am I supposed to do? A simple "Hi, looks like your Perkeep is empty for now! You might want to [upload] some files, [import] some stuff from other services or [set up your mobile device] to automatically store photos and more in Perkeep." would be very helpful.

Next, clicking on the main menu, the sheer number of options is a bit daunting and creates confusion:
- "New set", what's the difference with "upload"?
- "Import share", what's the difference with "importers", and why is "import share" not in "importers" then? etc.

I try "help", I'm taken to a austere looking page, different from what I was just seeing. Have I done something wrong?
This page also doesn't have a link to take me back to the normal Perkeep web UI, uh oh, I really must have clicked on the wrong thing.

Of course, you have to click your browser's back button a bunch of times or remove anything after `ui` in the URL and go back to the base URL, but even if you aren't phased by this, it's an annoying thing to have to do. A simple: "back to Perkeep home" button would help, until the unstyled pages can be styled to look like the rest of the UI.

The `server status` and importers page have the same issue.
`Mobile setup` doesn't have a way back, but doesn't have the austere (lack of style) of the others.
`About` creates a overlay on the main page. That's pretty nice, but inconsistent. Why doesn't mobile setup do that too then?
This leaves the user with a feeling of never knowing what to expect next when clicking around to explore Perkeep, an app that's supposed to be their *safe haven* for their private, intimate data.

## New set
This creates a new set and then tells you "No results found".
This is confusing, logical (the newly created set effectively does not contain anything) but unhelpful. This button seems superfluous at this point in time, until it can help users create a set of already existing objects by helping them pick them, or offer to upload them.

## Importers
As mentioned, there's no way back to te main UI, but furthermore once you are finished configuring (in my case the mastodon one) there's no clear confirmation that you are done and can safely go back to the main ui.

### Running the Mastodon importer
I was able to connect my Mastodon account to Perkeep, but it created 3 different "directories" (sets) on web ui's main page, with similar, but different names:
![screenshot_2019-01-20 https pk jaccu se](https://user-images.githubusercontent.com/1172256/51442262-794cc280-1cdb-11e9-97ef-717abc8d8a99.png)

The first directory is empty.
The second directory contains a directory with the same name as the first directory, and it's also empty.
The third one contains "lines of text". It's actually an object representing my Mastodon account itself: why then is it showed in the same way as the rest?

I rather like this idea of "everything is an object", but maybe at least put it automatically in a directory (a set) with a helpful name, to let users know this is a special object, not their statuses.
This is super confusing.

Furthermore, the importer didn't actually work, but didn't tell me. You have to go back into the account after you've created it to check, and then you see the error.

## Uploading
I was able to open the upload dialog and pick a file, but dialog then went from 0% to closing without confirming anything. Wait, did my file upload properly? Did the upload failed?
I tried to upload first a pdf (and tried it again) then a picture (jpg).

I couldn't see them, and tried a search, to no avail (see next section). 

The logs, however, show the following:
```
2019/01/20 16:43:38 Received blob [sha224-f1a76656064e77eb3d12094b4469fffcb5f693a67944841aadc15cf1; 566 bytes]
2019/01/20 16:43:38 Received blob [sha224-4e531069846fffa7b8914c9fc7abbf3af916bd8e5148c38fb4137ce2; 786 bytes]
2019/01/20 16:58:40 Received blob [sha224-868e3e1ac2fd3f1610afe2c0b3c65d15ca51b38e75dc78fea1b88a66; 669 bytes]
```
So the files have been uploaded. But where?

Eventually deleting the "Mastodon importer" revealed the picture I'd uploaded as a test. Why wasn't it shown before? The pdf is still nowhere to be found (even searching for "format:pdf").

### Searching
When I tried to search for the pdf I uploaded, simply typing "pdf" in the search, I was greeted with the folliwing message, shown in a somewhat menacing overlay with a red border:
> Invalid SearchQuery: Error parsing search expression "pdf": No support for literals yet at position 0, token: "pdf" See: https://perkeep.org/doc/search-ui

Did I do something wrong? I just wanted to find my file.

Lastly, there is no way "out of" a search that has failed. You just stay on an empty page, after you've closed the cryptic error message in a box with a red border.
A simple: "Nothing found. Go home?" on the screen would be helpful.

### Tags
I added a tag to the picture I uploaded, I got no confirmation it had worked.
When searching for the tag, I ran into the same error as for my other search.
Only when searching for "tag:mytag" did it find it.

## In conclusion
There are probably more usability issues, but this report is long enough as it is.
I hope you find it useful.
I'd be happy to help draft some help texts for the different pages, as I believe this can really make a difference for users not willing to bang their head against the app until they understand (like some of us will do).
A "first run" section in the documentation would also be very helpful, to guide new users in uploading their first pictures, importing content from 3rd party services and finding their content.

Hope this helps! 👋
